commit fa8419d08e58c82add921b4e4eb01eb589b8586b
Author: Nick Kossifidis <mick@madwifi-project.org>
Date:   Mon Feb 9 06:17:45 2009 +0200

    ath5k: Don't call reset on config_interface
    
     * We call reset too often and this can result various PHY problems.
      On config_interface we don't need to reset the whole device.
    
      TODO: Create a function hw_fast_reset that only resets the PCU
      (tx/rx stuff) and not the whole device so that we can use this for
      stuck beacons etc.
    
      Signed-off-by: Nick Kossifidis <mickflemm@gmail.com>
    
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/drivers/net/wireless/ath5k/base.c b/drivers/net/wireless/ath5k/base.c
index 3d36e84..6837ca9 100644
--- a/drivers/net/wireless/ath5k/base.c
+++ b/drivers/net/wireless/ath5k/base.c
@@ -2884,7 +2884,7 @@ ath5k_config_interface(struct ieee80211_hw *hw, struct ieee80211_vif *vif,
 {
 	struct ath5k_softc *sc = hw->priv;
 	struct ath5k_hw *ah = sc->ah;
-	int ret;
+	int ret = 0;
 
 	mutex_lock(&sc->lock);
 	if (sc->vif != vif) {
@@ -2910,9 +2910,7 @@ ath5k_config_interface(struct ieee80211_hw *hw, struct ieee80211_vif *vif,
 		}
 		ath5k_beacon_update(sc, beacon);
 	}
-	mutex_unlock(&sc->lock);
 
-	return ath5k_reset_wake(sc);
 unlock:
 	mutex_unlock(&sc->lock);
 	return ret;
